audformat/core/utils.py

```diff
@@ -550,17 +558,14 @@ def intersect(
     objs = _maybe_convert_single_level_multi_index(objs)
     _assert_index_alike(objs)
 
-    # sort objects by length
-    objs = sorted(objs, key=lambda obj: len(obj))
```

Comment:
So we are not sorting anymore? But wasn't it faster with sorting in your benchmark? If we sort, we should probably add another early stop criterion in case the first list is empty.

Reply:
Yes, it's faster with sorting, but sorting would fail the following test:

```python
        (
            [
                audformat.filewise_index(['f3', 'f2', 'f1']),
                audformat.filewise_index(['f1', 'f2']),
            ],
            audformat.filewise_index(['f2', 'f1']),
        ),
```
which we need to pass in order to support https://github.com/audeering/audformat/pull/254.

If we find a way to restore order at the end we can still use sort at the beginning, but so far I simply removed sorting which is still faster in most cases than the current implementation.

Reply:
There is `ordered-set`: https://pypi.org/project/ordered-set/, but we probably don't want to introduce a dependency because of that.

Reply:
I have restored sorting.

At the end, I restore the original order by adding an additional 
```python
index = objs[0].intersection(index)
```
which requires that we convert first the `index` from a list to an index.
Maybe there is another way to directly apply the list to `objs[0]`?

Reply:
I also tried to resort the list before creating the index with `_alike_index()` by
```python
index = [idx for idx in objs[0] if idx in index]
```
but interestingly this was up to 100x slower.

Reply:
Great we now return in order of the first object, can't think of a better solution.